### ISSUE TITLE
[BUG] Multifile support: file names get "lower-cased"

### ISSUE THREAD

**Issue by arnemertz** [compiler-explorer/compiler-explorer#3011]
**Describe the bug**
With multifile support, when we include upper case letters in the header names (e.g. "Source.hpp"), the compiler does not find them. They are instead found with the corresponding lower case names ("source.hpp")

**To Reproduce**
Start a multifile project, add a file named "Source.hpp", include it in another file with `#include "Source.hpp"` (note the capital S in both cases) -> compilation fails.

Change only the include to `#include "source.hpp"` -> compilation succeeds

**Expected behavior**
When I name a header "Source.hpp", I should be able to `#include "Source.hpp"`, and I should _not_ be able to `#include "source.hpp"`

**Link**
https://godbolt.org/z/zeeKK5aKr

**Comment by supergrecko** [comment]
I'm not able to reproduce this issue, I've tried https://godbolt.org/z/aceYG1a8Y

I had a look at your link, and it appears that you've written `source.hpp` inside the `add_executable()` call, when the header is called Source.hpp

**Comment by partouf** [comment]
I wouldn't generally add header files to add_executable() either, it's more work and definitely not necessary, but maybe that's just me.

**Comment by arnemertz** [comment]
Yes, my bad, I overlooked that the header was in the add_executable() as well. I used a link Matt had posted on reddit as basis ;-)
Sorry for the false report
